### ISSUE TITLE
Strange behavior while erasing text

### ISSUE THREAD

**Issue by firatcetiner** [siqwin/mask_text_input_formatter#49]
I found a strange behavior while trying to erase the text inside a `TextFormField` or `TextField`. I tried this on both versions `1.2.1` and `2.0.0-nullsafety.2` and the behavior are the same. What am I missing?

### Demonstration

![](https://media.giphy.com/media/ZsHVcyUVEuS8Cxdp3V/giphy.gif)
### Widget and the formatter

```
TextField(
  keyboardType: TextInputType.number,
  inputFormatters: [
    MaskTextInputFormatter(
      mask: '0###-###-##-##',
      filter: {
        "#": RegExp(r'[0-9]'),
      },
   ),
 ],
),
```
#### flutter doctor -v
```
[✓] Flutter (Channel stable, 2.0.3, on Mac OS X 10.15.7 19H524 darwin-x64, locale en-TR)
    • Flutter version 2.0.3 at /Users/apple/SDK/flutter
    • Framework revision 4d7946a68d (4 days ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/apple/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (2 available)
    • MI 6 (mobile) • 16185061 • android-arm64  • Android 9 (API 28)
    • Chrome (web)  • chrome   • web-javascript • Google Chrome 89.0.4389.90

• No issues found!
```

**Comment by RakovskyiSerhii** [comment]
Hi, try set keyboardType to TextInputType.number. Works just fine for me

**Comment by firatcetiner** [comment]
> Hi, try set keyboardType to TextInputType.number. Works just fine for me

@RakovskyiSerhii The example widget is already does that.

**Comment by shanilsoni** [comment]
I am also facing this same issue even though keyboardType is set to 'TextInputType.number'.

**Comment by igorbaranovmd** [comment]
Have same issue, but only on android(with both input types), on iOS works fine

**Comment by thide11** [comment]
Hi, i have the same issue, remove the keyboardType works temporarily(until the fix) for me

**Comment by gabrielcpaz** [comment]
Same issue

**Comment by bhanuka96** [comment]
SAME ISSUE. This is very critical bug

**Comment by Mohamed50** [comment]
same with me any updates about issue?

**Comment by newcaiosantos** [comment]
SAME ISSUE. This is very critical bug

**Comment by joseph-fox** [comment]
I'm having the same issue.

**Comment by vizhan-lanars** [comment]
I'm having the same issue. 
The `keyboardType = TextInputType.text` can solve that, but it's not an option.

**Comment by alexvip** [comment]
same issue here!! how to fix it?

**Comment by wilsondefreitas-dev** [comment]
same.

**Comment by maxim-gapeenko-idf** [comment]
same

**Comment by maxim-gapeenko-idf** [comment]
this bug is related to flutter, not this library. Fixed by switching from stable to latest flutter master

**Comment by PauloOyama** [comment]
To fix this problem i used the command "flutter downgrade" and do the "flutter pub get" command, if there's some problem with package version  just adjust this package to some older version

**Comment by joaovvrodrigues** [comment]
@PauloOyama what version of Flutter did you need to be in to prevent this problem from occurring?

**Comment by PauloOyama** [comment]
Flutter Version 1.22.5
Flutter channel stable

**Comment by ZNoct** [comment]
I've the same issue too for iOS it's working fine

**Comment by apocalipse9** [comment]
Same problem here, just Android, IOS it's ok.

**Comment by joaovvrodrigues** [comment]
If you changed the flutter to Channel Beta this problem will not occur, if it is urgent, as it was for me, do this upgrade that will be solved.
The problem is not in the package, so do this upgrade and follow the updates for the flutter stable: https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel

**Comment by EndersonSS** [comment]
Alternatively use the package brasil_fields: ^1.1.1  in null safety version

**Comment by PauloOyama** [comment]
Another way that I found to fix this issue was to download some other keyboard like "GBoard" in PlayStore and then change the default keyboard in options, maybe the problem it's in Samsung default keyboard.

**Comment by bhanuka96** [comment]
> Flutter Version 1.22.5
> Flutter channel stable

I downgrade `2` to `1.22.6`. Now, it's working. @PauloOyama Is this issue fix on the flutter master or dev channel?

**Comment by tattivitorino** [comment]
We're facing the same problem here.. IOS is ok but Android (on a physical device) it only works with the default keyboard.. meaning not setting any keyboard at all.. 

```
[✓] Flutter (Channel stable, 2.0.3, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-BR)
    • Flutter version 2.0.3 at /Users/tattivitorino/Tools/flutter
    • Framework revision 4d7946a68d (6 weeks ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/tattivitorino/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Users/tattivitorino/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/tattivitorino/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.
```

**Comment by Cortrano** [comment]
Same issue on Android. Flutter 2.0.5

**Comment by Cortrano** [comment]
I found a solution in the network graph. You can try this guy's fork, works great with flutter 2.0+
https://github.com/daveols/mask_text_input_formatter/commit/43d6e9bc60076c5379c7b0321c678922b851f9cd

**Comment by vitoryugulis** [comment]
Hi guys. This comment in another issue helped me solve this problem with wrong text erasing behaviour: [here](https://github.com/siqwin/mask_text_input_formatter/issues/41#issuecomment-829000935).
I just changed to beta channel and the lib works fine. I think it works on beta channel because of an issue on stable channel (as stated by siqwin in [here](https://github.com/flutter/flutter/issues/75502), and the fix is not on stable yet...

**Comment by chgsilva** [comment]
same problem here, any workarounds other than move to the dev channel? I am afraid of changing channels hence of the unexpected behaviors

**Comment by YuriOliveiraAbel** [comment]
same problem here

**Comment by firatcetiner** [comment]
This issue seems to be resolved with the latest stable Flutter version 2.2.0. So does #41.

**Comment by fhsbatista** [comment]
I tried on flutter 2.2.2 and the problem is no longer happening :)
